docs/root/intro/arch_overview/security/ssl.rst

```diff
@@ -213,6 +210,22 @@ A useful example handshaker, named ``SslHandshakerImplForTest``, lives in
 `this test <https://github.com/envoyproxy/envoy/blob/64bd6311bcc8f5b18ce44997ae22ff07ecccfe04/test/extensions/transport_sockets/tls/handshaker_test.cc#L174-L184>`_
 and demonstrates special-case ``SSL_ERROR`` handling and callbacks.
 
+.. _arch_overview_private_key_providers:
+
+Private key providers
+---------------------
+
+:ref:`Private key provider
+extensions<envoy_v3_api_msg_extensions.transport_sockets.tls.v3.PrivateKeyProvider>`
+implement BoringSSL private key methods (signing and decrypting). BoringSSL
+supports asynchronous handshakes with private key methods. This allows extending
+Envoy to support various key management schemes (such as TPM) and TLS
+acceleration. This mechanism uses `BoringSSL private key method interface
+<https://github.com/google/boringssl/blob/c0b4c72b6d4c6f4828a373ec454bd646390017d4/include/openssl/ssl.h#L1169>`_.
+:ref:`CryptoMb private key provider
```

Comment:
I think that's because those filters existed before we had `contrib`, so their docs are still in core docs. That should get cleaned up at some point. But new `contrib` extensions shouldn't add core docs.

Reply:
Ok, I reverted the docs change. I think we can re-add the docs when the contrib doc section is there.